cardano-db/src/Cardano/Db/Operations/Other/ConsumedTxOut.hs

```diff
@@ -0,0 +1,505 @@
+      pure False
+    -- the whole list is greater that maxTxInId
+    ([], ys) -> do
+      shouldCreateConsumedTxOut trce ranCreateConsumedTxOut
+      updatePageEntries txOutTableType ys
+      pure True
+    -- the list has both bellow and above maxTxInId
+    (xs, ys) -> do
+      deletePageEntries txOutTableType xs
+      shouldCreateConsumedTxOut trce ranCreateConsumedTxOut
+      updatePageEntries txOutTableType ys
+      pure True
+
+shouldCreateConsumedTxOut ::
+  (MonadIO m, MonadBaseControl IO m) =>
+  Trace IO Text ->
+  Bool ->
+  ReaderT SqlBackend m ()
+shouldCreateConsumedTxOut trce rcc =
+  unless rcc $ do
+    liftIO $ logInfo trce "Created ConsumedTxOut when handling page entries."
+    createConsumedIndexTxOut
+
+-- | Update
+updatePageEntries ::
+  MonadIO m =>
+  TxOutTableType ->
+  [ConsumedTriplet] ->
+  ReaderT SqlBackend m ()
+updatePageEntries txOutTableType = mapM_ (updateTxOutConsumedByTxIdUnique txOutTableType)
+
+updateTxOutConsumedByTxIdUnique :: MonadIO m => TxOutTableType -> ConsumedTriplet -> ReaderT SqlBackend m ()
+updateTxOutConsumedByTxIdUnique txOutTableType ConsumedTriplet {ctTxOutTxId, ctTxOutIndex, ctTxInTxId} =
+  case txOutTableType of
+    TxOutCore -> updateWhere [C.TxOutTxId ==. ctTxOutTxId, C.TxOutIndex ==. ctTxOutIndex] [C.TxOutConsumedByTxId =. Just ctTxInTxId]
+    TxOutVariantAddress -> updateWhere [V.TxOutTxId ==. ctTxOutTxId, V.TxOutIndex ==. ctTxOutIndex] [V.TxOutConsumedByTxId =. Just ctTxInTxId]
+
+-- this builds up a single delete query using the pageEntries list
+deletePageEntries ::
+  MonadIO m =>
+  TxOutTableType ->
+  [ConsumedTriplet] ->
+  ReaderT SqlBackend m ()
+deletePageEntries txOutTableType = mapM_ (\ConsumedTriplet {ctTxOutTxId, ctTxOutIndex} -> deleteTxOutConsumed txOutTableType ctTxOutTxId ctTxOutIndex)
+
+deleteTxOutConsumed :: MonadIO m => TxOutTableType -> TxId -> Word64 -> ReaderT SqlBackend m ()
+deleteTxOutConsumed txOutTableType txOutId index = case txOutTableType of
+  TxOutCore -> deleteWhere [C.TxOutTxId ==. txOutId, C.TxOutIndex ==. index]
+  TxOutVariantAddress -> deleteWhere [V.TxOutTxId ==. txOutId, V.TxOutIndex ==. index]
+
+--------------------------------------------------------------------------------------------------
+-- Raw Queries
+--------------------------------------------------------------------------------------------------
+
+createConsumedIndexTxOut ::
+  forall m.
+  ( MonadBaseControl IO m
+  , MonadIO m
+  ) =>
+  ReaderT SqlBackend m ()
+createConsumedIndexTxOut = do
+  handle exceptHandler $ rawExecute createIndex []
+  where
+    createIndex =
+      "CREATE INDEX IF NOT EXISTS idx_tx_out_consumed_by_tx_id ON tx_out (consumed_by_tx_id)"
+
+    exceptHandler :: SqlError -> ReaderT SqlBackend m a
+    exceptHandler e =
+      liftIO $ throwIO (DBPruneConsumed $ show e)
+
+createConsumedConstraintTxOut ::
+  forall m.
+  ( MonadBaseControl IO m
+  , MonadIO m
+  ) =>
+  ReaderT SqlBackend m ()
+createConsumedConstraintTxOut = do
```

Comment:
```suggestion
createPruneConstraintTxOut = do
```
since it's only necessary for prune. Also this is not currently used, see https://github.com/IntersectMBO/cardano-db-sync/pull/1820/files#r1757131157

Reply:
I was working through previous reviews and resolving conversations to the one's that I'd done. Not finished this yet :)